lib/error/serverSelection.js

```diff
@@ -34,13 +34,13 @@ class MongooseServerSelectionError extends MongooseError {
     // Special message for a case that is likely due to IP whitelisting issues.
     const isAtlasWhitelistError = isAtlas(reason) &&
       allServersUnknown(reason) &&
-      err.message.indexOf('bad auth') === -1 &&
-      err.message.indexOf('Authentication failed') === -1;
+      !err.message.includes('bad auth') && // David Xie: using String.includes() is more self-evident (compared to String.indexOf() === -1)
```

Comment:
I agree that `includes` provides better readability, but after we merge this, the comment won't mean much in the new context.

Nitpick: I think we can make it even more explicit to `err.message.includes('bad auth') === false` over `!err.message.includes('bad auth')` because it's much easier to not notice the `!` than the ` === false`, what do you think?